### ISSUE TITLE
Function formatting breaks source code by placeing a brace inside a comment

### ISSUE THREAD

**Issue by bruno-medeiros** [rust-lang-nursery/rustfmt#1113]
Running rustfmt 0.5.0 on this:

```
pub fn foo() -> fmt::Result 
//pub fn writeStringToken 
{
    panic!()
}
```

formats into this:

```
pub fn foo() -> fmt::Result //pub fn writeStringToken {
    panic!()
}
```

Which is invalid because it places the opening brace on the line-comment.

**Comment by zenhack** [comment]
I just hit this myself (literally the first time I ran rustfmt...). Another test case for what is probably the same bug: 

Before:

```rust
fn example(
        arg: i32, // a comment
    ) -> i32 {
    arg
}
```

After:

```rust


fn example(arg: i32 // a comment) -> i32 {
    arg
}
```

**Comment by topecongiro** [comment]
@zenhack The case of your example is fixed via #1596.
```rust
fn example(arg: i32 // a comment
           ) -> i32 {
    arg
}
```
There is room for improvement on how the code should be formatted, but at least rustfmt does not break the code.

**Comment by zenhack** [comment]
Ok, great, sorry for the noise -- hasn't hit a release yet :/
